app/javascript/packages/document-capture/components/in-person-location-step-po-search-beta.tsx

```diff
@@ -0,0 +1,212 @@
+import { useState, useEffect, useCallback, useRef, useContext } from 'react';
+import { useI18n } from '@18f/identity-react-i18n';
+import { PageHeading, SpinnerDots } from '@18f/identity-components';
+import { request } from '@18f/identity-request';
+import BackButton from './back-button';
+import LocationCollection from './location-collection';
+import LocationCollectionItem from './location-collection-item';
+import AnalyticsContext from '../context/analytics';
+import AddressSearch from './address-search';
+import InPersonContext from '../context/in-person';
+
+interface PostOffice {
+  address: string;
+  city: string;
+  name: string;
+  phone: string;
+  saturday_hours: string;
+  state: string;
+  sunday_hours: string;
+  weekday_hours: string;
+  zip_code_4: string;
+  zip_code_5: string;
+}
+
+interface FormattedLocation {
+  formattedCityStateZip: string;
+  id: number;
+  name: string;
+  phone: string;
+  saturdayHours: string;
+  streetAddress: string;
+  sundayHours: string;
+  weekdayHours: string;
+}
+interface LocationQuery {
+  streetAddress: string;
+  city: string;
+  state: string;
+  zipCode: string;
+}
+
+export const LOCATIONS_URL = '/verify/in_person/usps_locations';
+
+const getUspsLocations = (address) =>
+  request(LOCATIONS_URL, {
+    method: 'post',
+    json: { address },
+  });
+
+const formatLocation = (postOffices: PostOffice[]) => {
+  const formattedLocations = [] as FormattedLocation[];
+  postOffices.forEach((po: PostOffice, index) => {
+    const location = {
+      formattedCityStateZip: `${po.city}, ${po.state}, ${po.zip_code_5}-${po.zip_code_4}`,
+      id: index,
+      name: po.name,
+      phone: po.phone,
+      saturdayHours: po.saturday_hours,
+      streetAddress: po.address,
+      sundayHours: po.sunday_hours,
+      weekdayHours: po.weekday_hours,
+    } as FormattedLocation;
+    formattedLocations.push(location);
+  });
+  return formattedLocations;
+};
+
+const snakeCase = (value: string) =>
+  value
+    .split(/(?=[A-Z])/)
+    .join('_')
+    .toLowerCase();
+
+// snake case the keys of the location
+const prepToSend = (location: object) => {
+  const sendObject = {};
+  Object.keys(location).forEach((key) => {
+    sendObject[snakeCase(key)] = location[key];
+  });
+  return sendObject;
+};
+
+function InPersonLocationStepPoSearchBeta({ onChange, toPreviousStep }) {
```

Comment:
I don't know that we need to qualify this as "beta" ? I think it'd save us some future work (and risk of forgetting to follow-up) to leave it out. But in terms of suffixes, if this is a "step", I think it'd be good to follow convention of having the "step" suffix.

```suggestion
function InPersonLocationStepPOSearchStep({ onChange, toPreviousStep }) {
```

Reply:
I missed that the name already had another "step". Maybe we can simplify it further?

e.g.

```suggestion
function InPersonLocationSearchStep({ onChange, toPreviousStep }) {
```